### PR TITLE
Graphite: prefer more specific template over default template when possible

### PR DESCRIPTION
**Pull request by flisky**

### CODE REVIEW

**Comment by otoolep** [comment]
@jwilder

File: services/graphite/parser_test.go [hunk 1]
```diff
@@ -602,6 +602,27 @@ func TestApplyTemplateSpecific(t *testing.T) {
 	}
 }
 
+// Test that most specific template is N/A
```
Comment (otoolep) [inline services/graphite/parser_test.go]: Does this mean that "the most specific template is not applicable? not applied? doesn't matter?" The comment is a little ambiguous.

**Comment by otoolep** [comment]
@flisky -- have you signed the CLA?

File: services/graphite/parser_test.go (hunk 1 above)
Comment (flisky) [inline services/graphite/parser_test.go]: Ahem, the first match should be applied in this case. However, it goes deeper to find a more specific one, but cannot find it. The original result will roll back to the global default template, not the first match.

**Comment by flisky** [comment]
Yes. I'm the guy who contributes #4807.

**Comment by otoolep** [comment]
Ah yes, I remember that PR. Thanks.

**Comment by jwilder** [comment]
Thanks @flisky

**Comment by Shimi** [comment]
I have an issue with 0.12.1 where both specific filter and the default filter apply. Will it fix it?

**Pull request merged**